app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -439,45 +442,43 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     /**
      * Parses links from HTML string, and makes the links clickable in the specified TextView.<br>
-     * Uses {@link #makeLinkClickable(SpannableStringBuilder, URLSpan)}.
+     * Uses {@link //#makeLinkClickable(url,start,end,strBuilder)}.
      * @see <a href="https://stackoverflow.com/questions/12418279/android-textview-with-clickable-links-how-to-capture-clicks">Source</a>
      */
-    private void setTextViewHTML(TextView text, String html)
-    {
+
+    private void setTextViewHTML(TextView text, String html){
         CharSequence sequence = Html.fromHtml(html);
-        SpannableStringBuilder strBuilder = new SpannableStringBuilder(sequence);
+        SpannableString strBuilder = new SpannableString(sequence);
         URLSpan[] urls = strBuilder.getSpans(0, sequence.length(), URLSpan.class);
         for (URLSpan span : urls) {
-            makeLinkClickable(strBuilder, span);
+
+            int start = strBuilder.getSpanStart(span);
+            int end = strBuilder.getSpanEnd(span);
+            String url = span.getURL();
+
+            makeLinkClickable(url,start,end,strBuilder);
         }
         text.setText(strBuilder);
         text.setMovementMethod(LinkMovementMethod.getInstance());
     }
 
+
     /**
-     * Sets onClick handler to launch browser for the specified URLSpan.
-     * @see <a href="https://stackoverflow.com/questions/12418279/android-textview-with-clickable-links-how-to-capture-clicks">Source</a>
+     * URLSpan class extends ClickableSpan implements ParcelableSpan hance avoiding the memeory leak
+     *URLSpan  open the url, by launching an an Activity with an Intent#ACTION_VIEW intent.
+     * https://developer.android.com/reference/android/text/style/URLSpan
      */
```

Comment:
Can you please complete Javadoc by adding parameters?

Reply:
is it supposed to be like this

{*new URLSpan(url)  open the url, by launching an an Activity with an Intent#ACTION_VIEW intent.}

Reply:
> * @param  end        span send 

end maybe?

> * @param  strBuilder span send
This does not seem understandable to me